### PR TITLE
Add documentation for turning debugging on in REPL

### LINKED ISSUE #31741: Add docs for JULIA_DEBUG
Add JULIA_DEBUG to docs and also the 
JULIA_DEBUG environment variable section should be expanded to explain how to turn on the debug message in the REPL as well (cf https://docs.julialang.org/en/v1/stdlib/Logging/#Environment-variables-1)

### PR DESCRIPTION
**Pull request by tanhevg**
fix #31741

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
Yes, please. I've wanted to know how to do this since logging was added in 1.0.

**Comment by kshyatt** [comment]
This has been open for a few days -- any chance of a review, @c42f ?

**Review by c42f — COMMENTED** [review]
Thanks for this, and happy first pull request! I think this a helpful tutorial-like example to connect the use of the special `ENV` variable with the more terse description further up the page.

I suggested a rewording to emphasize that this enables debug messages specifically for functions defined in module `Main`. Does that seem clear to you?

**Comment by tanhevg** [comment]
@c42f , I have made some changes to the wording. I think it is important to stress in the text that module `Main` is used, and why it is used. In general, I agree that this bit of the documentation is a bit "terse". Probably the whole subsection needs rewriting; I might have a look at this as a separate PR.

**Comment by c42f** [comment]
Looks good to me, thanks.

I agree that the logging docs are too terse and could use a rework. I really like the following article describing technical documentation which I discovered relatively recently: https://www.divio.com/blog/documentation/. In those terms I think the logging docs are largely "explanation" and "reference", whereas they would be more useful to the average user if they had more "tutorial" and "how-to" style content.

I'm not sure there's a lot of precedent for that style of arrangement in the Julia docs as a whole, but we could arrange as subsections within the logging stdlib docs I guess.

File: stdlib/Logging/docs/src/index.md [hunk 1]
```diff
@@ -193,6 +193,24 @@ Similarly, the environment variable can be used to enable debug logging of
 modules, such as `Pkg`, or module roots (see [`Base.moduleroot`](@ref)). To
 enable all debug logging, use the special value `all`.
 
+To turn debug logging on from the REPL, set `ENV["JULIA_DEBUG"]` to the
+name of the module of interest. Functions defined in the REPL belong to
```
Comment (mbauman) [inline stdlib/Logging/docs/src/index.md]: This trailing whitespace broke CI on master.  https://github.com/JuliaLang/julia/pull/33641

Comment (c42f) [reply stdlib/Logging/docs/src/index.md]: Oh dear, looks like this was my mistake.

**Pull request merged**